src/publisher/api_v2_views.py

```diff
@@ -107,16 +110,18 @@ def article_version(request, id, version):
 
 @api_view(['POST'])
 def article_fragment(request, art_id, fragment_id):
-    only_published = is_authenticated(request)
+    if not is_authenticated(request):
+        return Response("not authenticated. only authenticated admin users can modify content", status=403)
     try:
         reserved_keys = [XML2JSON]
         ensure(fragment_id not in reserved_keys, "that key is taken")
         with transaction.atomic():
-            av = logic.most_recent_article_version(art_id, only_published)
-            data = request.data
-            frag, created, updated = fragment_logic.add(av.article, fragment_id, data, update=True)
-            ensure(created or updated, "fragment was not created/updated")
-            fragment_logic.merge_if_valid(av, quiet=False)
+            avs = logic.article_version_list(art_id, only_published=False)
+            for av in avs:
```

Comment:
This was not how fragments were intended to work

A fragment can be attached to an article or an articleversion. 
if a fragment exists at the article level, it gets applied to all article versions. 
if a fragment exists at the articleversion level, it only gets applied to that articleversion.

what you are now doing is adding the same fragment many times at the articleversion level.